builders/gradle.go

```diff
@@ -58,14 +58,20 @@ func (builder *GradleBuilder) Analyze(m module.Module, allowUnresolved bool) ([]
 	moduleConfigurationKey := strings.Split(m.Name, ":")
 	taskConfiguration := "compile"
 	taskName := moduleConfigurationKey[0]
-	if len(moduleConfigurationKey) == 2 {
+	if taskName != "" {
+		// this is not the root gradle project
+		taskName = taskName + ":"
+	}
+	if len(moduleConfigurationKey) == 2 && moduleConfigurationKey[1] != "" {
```

Comment:
Move this up to be next to the definition of `taskConfiguration`, so it's clear that your intent is to use a default value.